### PR TITLE
Add additional columns

### LINKED ISSUE #2: Tracking if a run uses the --full-refresh tag
We've been using our own version of audit logging, but are planning on moving to this. One thing that is really useful for our analysis is tracking if a run uses the `--full-refresh` flag or not. For various reasons we periodically do a full refresh overnight and want to be able to exclude those runs/model builds from some of the analysis. 

I think this should be able to be done by adding a column to the tables and amending the log_audit_event macro to take flags.FULL_REFRESH? Does this make sense as a thing to do? And, if it is, I presume that its necessary to write an additional macro that moves the old schema to the new one, or is that not the right way to approach it?

### LINKED ISSUE #10: Track additional information
Similar to #2, I'd like to see the following information in the event logging table:
- User
- Target

### PR DESCRIPTION
**Pull request by Limess**
This change automatically migrates schemas to add the new columns:

* `event_target_name` - the name of the DBT target used
* `event_user` - the name of the user who ran DBT
* `event_is_full_refresh` - whether the run was a full refresh

adding NULL values for these columns to existing rows.

I haven't made any changes to supplied dashboards (periscope or looker) but have updated the looker models with these changes (I haven't got any experience with LookML but they looked simple enough).

This conflicts with what https://github.com/fishtown-analytics/dbt-event-logging/pull/12/files is trying to do, but this can act as a stop-gap in the meantime for those who wish to make use of the additional events in auditing.

Closes https://github.com/fishtown-analytics/dbt-event-logging/issues/2
Closes https://github.com/fishtown-analytics/dbt-event-logging/issues/10

Example database rows after the tests are ran with this change:

```
| event_name                 | event_timestamp            | event_schema   | event_model               | event_user | event_target | event_is_full_refresh | invocation_id                        |
|----------------------------|----------------------------|----------------|---------------------------|------------|--------------|-----------------------|--------------------------------------|
| run started                | 2020-02-26 14:33:09.14614  |                |                           | postgres   | postgres     | t                     | c6b106a3-addd-4375-9b9f-3c68144f9728 |
| model deployment started   | 2020-02-26 14:33:09.303889 | analytics_meta | stg_dbt_audit_log         | postgres   | postgres     | t                     | c6b106a3-addd-4375-9b9f-3c68144f9728 |
| model deployment started   | 2020-02-26 14:33:09.351807 | analytics      | my_model                  | postgres   | postgres     | t                     | c6b106a3-addd-4375-9b9f-3c68144f9728 |
| model deployment completed | 2020-02-26 14:33:09.303889 | analytics_meta | stg_dbt_audit_log         | postgres   | postgres     | t                     | c6b106a3-addd-4375-9b9f-3c68144f9728 |
| model deployment completed | 2020-02-26 14:33:09.351807 | analytics      | my_model                  | postgres   | postgres     | t                     | c6b106a3-addd-4375-9b9f-3c68144f9728 |
| model deployment started   | 2020-02-26 14:33:09.584329 | analytics_meta | stg_dbt_model_deployments | postgres   | postgres     | t                     | c6b106a3-addd-4375-9b9f-3c68144f9728 |
| model deployment started   | 2020-02-26 14:33:09.60672  | analytics_meta | stg_dbt_deployments       | postgres   | postgres     | t                     | c6b106a3-addd-4375-9b9f-3c68144f9728 |
| model deployment completed | 2020-02-26 14:33:09.584329 | analytics_meta | stg_dbt_model_deployments | postgres   | postgres     | t                     | c6b106a3-addd-4375-9b9f-3c68144f9728 |
| model deployment completed | 2020-02-26 14:33:09.60672  | analytics_meta | stg_dbt_deployments       | postgres   | postgres     | t                     | c6b106a3-addd-4375-9b9f-3c68144f9728 |
| run completed              | 2020-02-26 14:33:09.815671 |                |                           | postgres   | postgres     | t                     | c6b106a3-addd-4375-9b9f-3c68144f9728 |
| model deployment started   | 2020-02-26 14:33:12.392849 | analytics_meta | stg_dbt_audit_log         | postgres   | postgres     | f                     | 60eaa89e-8545-402b-85c0-b1afacecf036 |
| model deployment started   | 2020-02-26 14:33:12.399048 | analytics      | my_model                  | postgres   | postgres     | f                     | 60eaa89e-8545-402b-85c0-b1afacecf036 |
| model deployment completed | 2020-02-26 14:33:12.392849 | analytics_meta | stg_dbt_audit_log         | postgres   | postgres     | f                     | 60eaa89e-8545-402b-85c0-b1afacecf036 |
| model deployment completed | 2020-02-26 14:33:12.399048 | analytics      | my_model                  | postgres   | postgres     | f                     | 60eaa89e-8545-402b-85c0-b1afacecf036 |
| model deployment started   | 2020-02-26 14:33:12.591672 | analytics_meta | stg_dbt_deployments       | postgres   | postgres     | f                     | 60eaa89e-8545-402b-85c0-b1afacecf036 |
| model deployment completed | 2020-02-26 14:33:12.591672 | analytics_meta | stg_dbt_deployments       | postgres   | postgres     | f                     | 60eaa89e-8545-402b-85c0-b1afacecf036 |
| model deployment started   | 2020-02-26 14:33:12.61017  | analytics_meta | stg_dbt_model_deployments | postgres   | postgres     | f                     | 60eaa89e-8545-402b-85c0-b1afacecf036 |
| model deployment completed | 2020-02-26 14:33:12.61017  | analytics_meta | stg_dbt_model_deployments | postgres   | postgres     | f                     | 60eaa89e-8545-402b-85c0-b1afacecf036 |
| run completed              | 2020-02-26 14:33:12.808472 |                |                           | postgres   | postgres     | f                     | 60eaa89e-8545-402b-85c0-b1afacecf036 |
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -67,6 +77,11 @@ jobs:
             dbt run --target snowflake --full-refresh
             dbt run --target snowflake
 
+            dbt run-operation drop_audit_schema --target snowflake
+            dbt run-operation create_legacy_audit_table --target postgres
```
Comment (clrcrl) [inline .circleci/config.yml]: I think this should be `--target snowflake` (and below too!)

File: README.md [hunk 2]
```diff
@@ -33,5 +33,18 @@ That's it! You'll now have a stream of events for all dbt invocations in your
 warehouse.
 
 ### Adapter support
+
 This package is currently compatible with dbt's Snowflake, Redshift, and
 Postgres integrations.
+
+### Migration guide
+
+#### v0.1.17 -> UNRELEASED
```
Comment (clrcrl) [inline README.md]: We can name this `v0.2.0`

**Review by clrcrl — CHANGES_REQUESTED** [review]
Wow! Thank you so much for this PR! 🙌

Just kicked off the tests, but I think they might fail! See related comments :)

**Comment by clrcrl** [comment]
Lol never mind, we haven't set up the CircleCI side of these tests yet -- just a minute!

File: .circleci/config.yml [hunk 3]
```diff
@@ -56,6 +61,11 @@ jobs:
             dbt run --target redshift --full-refresh
             dbt run --target redshift
 
+            dbt run-operation drop_audit_schema --target redshift
+            dbt run-operation create_legacy_audit_table --target postgres
```
Comment (clrcrl) [inline .circleci/config.yml]:
```
-- target redshift
```
(and below!)

File: macros/audit.sql [hunk 4]
```diff
@@ -46,39 +52,75 @@
...
+
+
+    {% if audit_table_exists -%}
+
+        {%- set columns_to_create = [] -%}
+
+        {%- set existing_columns = adapter.get_columns_in_relation(audit_table)|map(attribute='column')|list -%}
+
+        {%- for required_column in required_columns -%}
+            {%- if required_column[0] not in existing_columns -%}
```
Comment (clrcrl) [inline macros/audit.sql:81]: For Snowflake reasons:
```suggestion
            {%- if required_column[0] not in existing_columns | lower -%}
```

**Review by clrcrl — CHANGES_REQUESTED** [review]

File: macros/audit.sql (hunk 4 above)
Comment (clrcrl) [reply macros/audit.sql:81]: Basically, Snowflake returns column  names as upper case strings. It's silly. But for that reason, `'event_name' != 'EVENT_NAME'`

Comment (Limess) [reply macros/audit.sql:81]: tah, didn't know that! Added the `lower `when getting existing_columns

**Pull request merged**